packages/truffle-contract/lib/execute.js

```diff
@@ -145,10 +192,21 @@ var execute = {
         execute
           .getGasEstimate
           .call(constructor, params, network.blockLimit)
-          .then(gas => {
-            params.gas = gas
+          .then(result => {
+            (result.error)
+              ? context.reason = result.error.reason
+              : context.reason = null;
+
```

Comment:
Is this getting attached at the receipt handler? Check

Reply:
Yes it is - we also have tests for reason string handling across a matrix of the following:
+ `deploy` and `send`  
+ both states of vmErrorsOnResponse
+ non-existence of reason / geth 
+ ganache as provider and server (the former at truffle-contract / the latter at scenario)